validation/src/lib.rs

```diff
@@ -569,23 +569,24 @@ impl<C, N, P, SC, TxPool, B> consensus::Environment<Block> for ProposerFactory<C
 	// Rust bug: https://github.com/rust-lang/rust/issues/24159
 	sp_api::StateBackendFor<P, Block>: sp_api::StateBackend<HasherFor<Block>> + Send,
 {
+	type CreateProposer = Pin<Box<
```

Comment:
Is it necessary for this to be `Pin`? In case the future is `Unpin` I would suggest pinning on `poll`ing.

Reply:
I guess not - I'll avoid pinning in the follow-up if possible.